lib/posix/options/grp.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/posix/grp.h>
+
+#ifdef CONFIG_POSIX_THREAD_SAFE_FUNCTIONS
+
+int getgrnam_r(const char *name, struct group *grp, char *buffer, size_t bufsize,
+	       struct group **result)
+{
+	ARG_UNUSED(name);
+	ARG_UNUSED(grp);
+	ARG_UNUSED(buffer);
+	ARG_UNUSED(bufsize);
+	ARG_UNUSED(result);
+
+	return ENOSYS;
```

Comment:
Isn't this supposed to be as follows? (ditto elsewhere in the patch... and potentially at least one other place in-tree in lib/posix it would seem)


```suggestion
	errno = ENOSYS;
	return -1;
```

Reply:
Thanks for the review, I had a look at the spec again, the current implementation is correct:

https://pubs.opengroup.org/onlinepubs/9699919799/functions/getgrnam_r.html

> The getgrnam_r() function shall return zero on success or if the requested entry was not found and no error has occurred. If any error has occurred, an error number shall be returned to indicate the error.

I also cross-checked this function with the implementation in nuttx ([1](https://github.com/apache/nuttx/blob/master/libs/libc/grp/lib_getgrnamr.c#L85), [2](https://github.com/apache/nuttx/blob/master/libs/libc/grp/lib_getgrbufr.c#L86)).

For implementation that requires `errno = E*`, the spec would mention:
> If the aio_error() function fails, it shall return -1 and set errno to indicate the error.

which is the case for [`aio_error()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/aio_error.html)